### ISSUE TITLE
flyimport / auto-import assist does not suggest structs defined in macros

### ISSUE THREAD

**Issue by jhgg** [rust-analyzer/rust-analyzer#10764]
With the following code:

```rust
mod foo {
    pub struct NotDefinedInMacro;

    macro_rules! define_struct {
        () => {
            pub struct DefinedInMacro;
        };
    }

    define_struct!();
}

fn main() {
    let x = NotDefine$0dInMacro {};
    let y = DefinedIn$1Macro {};
    let z = Defi$2;
}
```

In `$0` the `import foo::NotDefinedInMacro` assist is available.
In `$1` the `import foo::DefinedInMacro` assist is **not** available.
In `$2` the fly-import for `DefinedInMacro` is **not** available.

## video:

![PtpYwqFUuK](https://user-images.githubusercontent.com/5489149/141677989-b5fd4764-1c69-443a-bbce-d52a36c47934.gif)

This bug is actually quite annoying as we use a lot of macros to generate structs, which no longer get the assist/fly import :(

**Comment by Veykril** [comment]
I think this is due to our index for these things still being source file based which means we don't yet look into macros, cc https://github.com/rust-analyzer/rust-analyzer/issues/4842

**Comment by lnicola** [comment]
Fixed in #10872
